api/src/main/java/io/grpc/MetricSink.java

```diff
@@ -99,5 +99,30 @@ default void recordLongHistogram(LongHistogramMetricInstrument metricInstrument,
       List<String> requiredLabelValues, List<String> optionalLabelValues) {
   }
 
+  /**
+   * Record a long gauge value.
+   *
+   * @param value The value to record.
+   * @param requiredLabelValues A list of required label values for the metric.
+   * @param optionalLabelValues A list of additional, optional label values for the metric.
+   */
+  default void recordLongGauge(LongGaugeMetricInstrument metricInstrument, long value,
+      List<String> requiredLabelValues, List<String> optionalLabelValues){
+  }
+
+  /**
+   * Registers a callback to produce metric values for only the listed instruments. The returned
+   * registration must be closed when no longer needed, which will remove the callback.
```

Comment:
Is closing the registration the job of the LB that is producing the metrics?

Reply:
Yeah. During LB.shutdown() it'd want to close the registration to allow itself to be GCed.